File: advertisingonrails/app/models/targetingprofile.rb

class TargetingProfile < ApplicationRecord


  has_many :AudienceSegments, class_name: 'AudienceSegment'
  has_many :GeoRegions, class_name: 'GeoRegion'
  has_many :ContentCategories, class_name: 'ContentCategory'
  has_many :BrandSafetyPolicy, class_name: 'BrandSafetyPolicy'
  has_many :DeviceCriteria, class_name: 'DeviceCriterion'

end

File: advertisingonaspdotnet/Domain/TargetingProfile.cs

using advertisingonaspdotnet.Contracts;

namespace advertisingonaspdotnet.Domain;

public class TargetingProfile
{
    public Guid Id { get; set; } = Guid.NewGuid();

 public virtual long? TargetingprofileId { get; set; } 
 public virtual string? Name { get; set; } 
public virtual ICollection<AudienceSegment> AudienceSegments { get; set; } = new List<AudienceSegment>();
public virtual ICollection<GeoRegion> GeoRegions { get; set; } = new List<GeoRegion>();
public virtual ICollection<ContentCategory> ContentCategories { get; set; } = new List<ContentCategory>();
public virtual BrandSafetyPolicy? BrandSafetyPolicy { get; set; } 
public virtual ICollection<DeviceCriterion> DeviceCriteria { get; set; } = new List<DeviceCriterion>();

    public static TargetingProfile FromRequest(TargetingProfileRequest request) {
        return new TargetingProfile {
            Id = request.Id,
            Name = request.Name,
        };
    }
}
